ui-modules/blueprint-composer/app/components/dsl-viewer/dsl-viewer.js

```diff
@@ -42,18 +42,24 @@ export function dslViewerDirective() {
     };
 
     function link(scope) {
-        scope.isTargetDsl = (dsl) => {
-            return dsl.kind === KIND.TARGET;
+        
+        scope.getIcon = (dsl) => {
```

Comment:
The template now uses `ng-switch` so there is no point of having this function. The class can be directly put into the HTML element. Less code for same result, yay 🎉

Reply:
I don't follow.  the `ng-switch` is on the mode, but the way the code works (unchanged) is that the icon is based on the name, not the mode.  So I think we still need an icon lookup somewhere.  I think JS is better than doing it in HTML.

Reply:
Good point regarding the mode vs name. However, I strongly disagree about doing this in Javascript. This is not the angular way because of performances. Calling the method in `ng-class` will force angular to compute this data on every digest cycle, whereas doing it in the HTML will trigger the evaluation only if the DSL name changes.

Reply:
good point; now uses `scope.$watch`; also a few other fixes